<a href="https://colab.research.google.com/github/Amal2802/Project2_readme/blob/main/Project2_part1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# Mount google drive
from google.colab import drive
drive.mount('/content/drive', force_remount=True)

Mounted at /content/drive


In [3]:
#Import
import pandas as pd
import numpy as np

In [7]:
# Load in the data
df_car = pd.read_csv('/content/Car_Insurance_Claim.csv.zip')
# Take a look at the dataframe
df_car

ID    AGE  GENDER      RACE DRIVING_EXPERIENCE    EDUCATION  \
0     569520    65+  female  majority               0-9y  high school   
1     750365  16-25    male  majority               0-9y         none   
2     199901  16-25  female  majority               0-9y  high school   
3     478866  16-25    male  majority               0-9y   university   
4     731664  26-39    male  majority             10-19y         none   
...      ...    ...     ...       ...                ...          ...   
9995  323164  26-39  female  majority             10-19y   university   
9996  910346  26-39  female  majority             10-19y         none   
9997  468409  26-39    male  majority               0-9y  high school   
9998  903459  26-39  female  majority             10-19y  high school   
9999  442696  26-39  female  majority               0-9y         none   

             INCOME  CREDIT_SCORE  VEHICLE_OWNERSHIP VEHICLE_YEAR  MARRIED  \
0       upper class      0.629027                1.0   after 2015      0.0   
1           poverty      0.357757                0.0  before 2015      0.0   
2     working class      0.493146                1.0  before 2015      0.0   
3     working class      0.206013                1.0  before 2015      0.0   
4     working class      0.388366                1.0  before 2015      0.0   
...             ...           ...                ...          ...      ...   
9995    upper class      0.582787                1.0  before 2015      0.0   
9996   middle class      0.522231                1.0   after 2015      0.0   
9997   middle class      0.470940                1.0  before 2015      0.0   
9998        poverty      0.364185                0.0  before 2015      0.0   
9999  working class      0.435225                1.0  before 2015      1.0   

      CHILDREN  POSTAL_CODE  ANNUAL_MILEAGE VEHICLE_TYPE  SPEEDING_VIOLATIONS  \
0          1.0        10238         12000.0        sedan                    0   
1          0.0        10238         16000.0        sedan                    0   
2          0.0        10238         11000.0        sedan                    0   
3          1.0        32765         11000.0        sedan                    0   
4          0.0        32765         12000.0        sedan                    2   
...        ...          ...             ...          ...                  ...   
9995       0.0        10238         16000.0        sedan                    0   
9996       1.0        32765             NaN        sedan                    1   
9997       1.0        10238         14000.0        sedan                    0   
9998       1.0        10238         13000.0        sedan                    2   
9999       1.0        10238         13000.0        sedan                    0   

      DUIS  PAST_ACCIDENTS  OUTCOME  
0        0               0      0.0  
1        0               0      1.0  
2        0               0      0.0  
3        0               0      0.0  
4        0               1      1.0  
...    ...             ...      ...  
9995     0               1      0.0  
9996     0               0      0.0  
9997     0               0      0.0  
9998     0               1      1.0  
9999     0               0      0.0  

[10000 rows x 19 columns]

#1. Source of data

https://www.kaggle.com/datasets/sagnik1511/car-insurance-data


#2. Brief description of data


it's annual auto insurance data and we're going to find out the actual behaviors of customers from the data.

#3. What is the target?

OUTCOME

#4. What does one row represent? (A person? A business? An event? A product?)

Business

#5. Is this a classification or regression problem?

This is a classification

#6. How many features does the data have?


#7. How many rows are in the dataset?

10000 rows

#8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

I need to uncover real customer behaviors from the data.


In [8]:
# Load in the data
df_healthcare = pd.read_csv('/content/healthcare-dataset-stroke-data.csv')
# Take a look at the dataframe
df_healthcare

id  gender   age  hypertension  heart_disease ever_married  \
0      9046    Male  67.0             0              1          Yes   
1     51676  Female  61.0             0              0          Yes   
2     31112    Male  80.0             0              1          Yes   
3     60182  Female  49.0             0              0          Yes   
4      1665  Female  79.0             1              0          Yes   
...     ...     ...   ...           ...            ...          ...   
5105  18234  Female  80.0             1              0          Yes   
5106  44873  Female  81.0             0              0          Yes   
5107  19723  Female  35.0             0              0          Yes   
5108  37544    Male  51.0             0              0          Yes   
5109  44679  Female  44.0             0              0          Yes   

          work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0           Private          Urban             228.69  36.6  formerly smoked   
1     Self-employed          Rural             202.21   NaN     never smoked   
2           Private          Rural             105.92  32.5     never smoked   
3           Private          Urban             171.23  34.4           smokes   
4     Self-employed          Rural             174.12  24.0     never smoked   
...             ...            ...                ...   ...              ...   
5105        Private          Urban              83.75   NaN     never smoked   
5106  Self-employed          Urban             125.20  40.0     never smoked   
5107  Self-employed          Rural              82.99  30.6     never smoked   
5108        Private          Rural             166.29  25.6  formerly smoked   
5109       Govt_job          Urban              85.28  26.2          Unknown   

      stroke  
0          1  
1          1  
2          1  
3          1  
4          1  
...      ...  
5105       0  
5106       0  
5107       0  
5108       0  
5109       0  

[5110 rows x 12 columns]

#1. Source of data

https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset


#2. Brief description of data

This dataset is used to predict whether a patient is likely to get stroke based on the input parameters like gender, age, various diseases, and smoking status.

#3. What is the target?

stroke

#4. What does one row represent? (A person? A business? An event? A product?)

A person

#5. Is this a classification or regression problem?

This is a classification

#6. How many features does the data have?


#7. How many rows are in the dataset?

5110 rows




